packages/core/test/publisher.spec.ts

```diff
@@ -23,24 +28,130 @@ describe("Publisher", () => {
   });
 
   describe("init", () => {
-    it("registers event listeners", async () => {
+    it("should process queue", async () => {
       const opts = { ttl: 1, prompt: true, relay: { protocol: "irn" }, tag: 0 };
-      const itemA = { topic: generateRandomBytes32(), message: "itemA", opts };
-      const itemB = { topic: generateRandomBytes32(), message: "itemB", opts };
+      const items = [
+        {
+          topic: generateRandomBytes32(),
+          message: "itemA",
+          opts: { ...opts, id: getId() },
+        },
+        {
+          topic: generateRandomBytes32(),
+          message: "itemB",
+          opts: { ...opts, id: getId() },
+        },
+        {
+          topic: generateRandomBytes32(),
+          message: "itemC",
+          opts: { ...opts, id: getId() },
+        },
+        {
+          topic: generateRandomBytes32(),
+          message: "itemD",
+          opts: { ...opts, id: getId() },
+        },
+        {
+          topic: generateRandomBytes32(),
+          message: "itemE",
+          opts: { ...opts, id: getId() },
+        },
+      ];
+
       const requestSpy = Sinon.spy();
       publisher.relayer.request = requestSpy;
+
       // Manually set some items in the queue.
-      publisher.queue.set(hashMessage(itemA.message), itemA);
-      publisher.queue.set(hashMessage(itemB.message), itemB);
-      expect(publisher.queue.size).to.equal(2);
+      items.forEach((item) => publisher.queue.set(item.opts.id.toString(), item));
+      expect(publisher.queue.size).to.equal(items.length);
+      // Emit heartbeat pulse event
+      publisher.relayer.core.heartbeat.events.emit(HEARTBEAT_EVENTS.pulse);
+
+      // Emit ACKs
+      items.forEach((item) =>
+        core.relayer.events.emit(RELAYER_EVENTS.message_ack, { id: item.opts.id }),
+      );
+
+      // -> Queue should clear after the ACKs.
+      expect(publisher.queue.size).to.equal(0);
+
+      // emit multiple pulses to ensure queue is cleared
+      await throttle(100);
```

Comment:
Why throttle before the first heartbeat pulse here rather than after?

Reply:
left over from different iterations of the test. Its redundant and I removed it 💯